components/OutivityForm/index.js

```diff
@@ -1,41 +1,90 @@
+  outivity,
+  onEditOutivity,
+  createOutivity,
+  isEdit,
+  selectedImage,
+  setSelectedImage,
+}) {
+  const router = useRouter();
   const inputRef = useRef(null);
-
+  const [message, setMessage] = useState("");
+  const handleCancel = () => {
+    const confirmed = window.confirm("Are you sure you want to cancel?");
+    if (confirmed) {
+      router.push("/");
+    }
+  };
   useEffect(() => {
     if (inputRef.current) {
       inputRef.current.focus();
     }
   }, []);
+  const handleImageChange = (event) => {
+    const file = event.target.files[0];
+    setSelectedImage(URL.createObjectURL(file));
+    setMessage(event.target.value);
+  };
+  const handleUnsetImagePreview = () => {
+    setMessage("");
+    setSelectedImage("");
+  };
 
   return (
     <main>
-      <StyledNewOutivitiesForm onSubmit={createOutivity}>
-        <h1>New Outivity</h1>
+      <StyledNewOutivitiesForm
+        onSubmit={isEdit ? onEditOutivity : createOutivity}
+      >
+        <h1>{isEdit ? "Edit Outivity" : "New Outivity"}</h1>
         <StyledNewOutivitiesFormFields>
           <StyledNewOutivitiesFormField>
-            <label htmlFor="outivityImage">Image</label>
-
+            <label htmlFor="outivityName">Outivity</label>
             <StyledNewOutivitiesFormInput
               type="text"
-              name="outivityImage"
-              id="outivityImage"
-              placeholder="Insert a url from unsplash..."
+              name="outivityName"
+              id="outivityName"
+              placeholder="Type in a name..."
               required
               autoFocus
               ref={inputRef}
+              defaultValue={isEdit ? outivity.title ?? "" : null}
             />
           </StyledNewOutivitiesFormField>
           <StyledNewOutivitiesFormField>
-            <label htmlFor="outivityName">Outivity</label>
+            <label htmlFor="outivityImage">Image</label>
+            {selectedImage && (
+              <>
+                <StyledFormPreviewImage
+                  alt="imagePreview"
+                  name="imagePreview"
+                  width={300}
+                  height={200}
+                  src={selectedImage}
+                />
+                <StyledFormPreviewImageUrl>
+                  {outivity?.image}
+                </StyledFormPreviewImageUrl>
+                <StyledFormRemoveButton
+                  type="button"
+                  onClick={handleUnsetImagePreview}
+                >
+                  Remove
+                </StyledFormRemoveButton>
+              </>
+            )}
             <StyledNewOutivitiesFormInput
-              type="text"
-              name="outivityName"
-              id="outivityName"
-              placeholder="Type in a name..."
+              type="file"
+              class="inputfile"
+              name="outivityImage"
+              id="outivityImage"
+              accept=".png, .jpeg, .jpg, .webp"
               required
```

Comment:
Hey, sorry to not approve you PR again, but we think you should only make the image required, when creating a new Outivity. But when editing, it shouldn't be required so I can just leave the image, that I already uploaded before. We made a suggestion to implement that: 
```suggestion
              required={!isEdit}
```

Reply:
We actually recognized this ourselves and already changed it.
Thank's for the hint!